File: cpp/ecbenchmark/src/operator/Permuted.cpp
#include "ecbenchmark/operator/Permuted.h"

namespace ecb {

    Permuted::Permuted(const std::vector<int>& permutation, Function* f)
    : Operator(f), _permutation(permutation) {

    }
    
    void Permuted::setPermutation(const std::vector<int>& permutation){
        this->_permutation = permutation;
    }
    
    std::vector<int> Permuted::permutation() const {
        return this->_permutation;
    }
    
    scalar Permuted::f(const std::vector<scalar>& x){
        std::vector<scalar> permuted_x;
        permuted_x.reserve(x.size());
        for (size_t i = 0 ; i  < x.size() ; ++i){
            permuted_x.push_back(x[_permutation[i]]);
        }
        return _innerFunction->f(permuted_x);
    }
}

File: java/ecbenchmark/src/ecbenchmark/operator/Permuted.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.operator;

import ecbenchmark.Operator;
import ecbenchmark.Function;

/**
 *
 * @author jcrada
 */
public class Permuted extends Operator {

    protected int[] permutation;

    public Permuted(int[] permutation, Function f) {
        super(f);
        this.permutation = permutation;
    }

    @Override
    public double f(final double[] x) {
        double[] permuted_x = new double[x.length];
        for (int i = 0; i < permutation.length; ++i) {
            permuted_x[i] = x[permutation[i]];
        }
        return innerFunction.f(permuted_x);
    }
}
